src/DynamoCore/Graph/Notes/NoteModel.cs

```diff
@@ -11,6 +11,16 @@ namespace Dynamo.Graph.Notes
     /// </summary>
     public class NoteModel : ModelBase
     {
+        /// <summary>
+        /// This action is triggered when undo command is pressed and a node is pinned
+        /// </summary>
+        internal event Action<ModelBase> UndoRequest;
+
+        /// <summary>
+        /// Checks if the node has just been pinned
+        /// </summary>
+        internal bool CreatedPinNode { get; set; }
```

Comment:
Also if we have to use this new boolean, it might be good to follow the code convention to start the bool with `Is`, e.g. `IsNodeJustPinned`